### DISCUSSION TITLE
Cast a recipe to a google cast device

### DISCUSSION THREAD

**Discussion by felixschndr** [mealie-recipes/mealie#3628] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Many people (including me) have a smart display in their kitchen. It'd be great if it was possible to show a recipe from mealie on such a device.

### Please provide a concise description of the feature that would resolve your issue.

Example: Nest Hubs (by Google) have the ability to display recipes. My idea would be to able to show recipes on such devices.

I was thinking about how to implement this. I think there could be much copied from Homeassistant. HA supports casting a screen to a smart display: https://www.home-assistant.io/integrations/cast/

Ideally the mealie instance would scan for cast devices, the webapp (phone or desktop) would let the user chose a device. Afterwards the mealie server would cast the recipe to the smart display.

### Please consider and list out some caveats or tradeoffs made in your design decision

*Could* take *a little bit* of fiddling. 

Since I have such smart displays I would be happily to test and debug this in collaboration with a developer.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [X] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
I think this would be pretty cool to add to Mealie. No idea if it's possible with our setup (or even where to begin)

**Comment by Mraedis** [comment]
Can you not do this already with a "Mealie Dashboard" + cast dashboard to device?

**Reply by felixschndr** [reply]
You mean with the help of HomeAssistant?

**Reply by Mraedis** [reply]
Yes, because you linked HA I assume you already have this installed, you can add mealie as a custom dashboard. I did not yet try the casting yet though.

**Reply by felixschndr** [reply]
I don't have Homeassistant installed at the moment. I just saw that HA is able to do it.
However I will try to use HA when I got some time

**Reply by Mraedis** [reply]
If you need to install HA to do it then that'd be a big waste of resources. HA is nice but a big thing to just cast a page. :) You could simply cast the screen from any other android device, no?

**Reply by felixschndr** [reply]
True... 

> You could simply cast the screen from any other android device, no?

That should work, will try this later when I am at home

**Reply by felixschndr** [reply]
I tried streaming my screen from my phone to the display with Mealie opened. That worked as expected. However it only streams the content of the screen. Interaction from the display (e.g. scrolling) is not possible.

Since this uses the Google Cast protocol and I assume HA is also just using Google Cast it is not possible to overcome this problem.

**Comment by felixschndr** [comment]
See https://github.com/mealie-recipes/mealie/discussions/3628#discussioncomment-9512338

**Comment by skank01** [comment]
Im looking for this too.
I come from a long way.
I use mealie as an addon in HA
First thing to do , before able to cast it, it had to be https
Mealie in home assistant is http

So i had to create reverse proxy using nginx manager... Then create a subdomain in cloudflare, and setup a reverse proxy in nginx manager + ssl certificate
Then i had to open 2 ports on my unifi setup
Now i can reach it from https
Next i created a button in home assistant to start a script where i start casting to a google home hub in my kitchen
This works
I can now see mealie on my hub, + i can even scroll in mealie
However
Before getting in mealie, i need to log in mealie
And there im stuck.
Cause when i set my cursor in my user field (or even password field) , i dont get a keyboard to type, so i cant log in
Stuck here
Really looking forward for a solution for this

Casting from phone is not a good idea cause you cant scroll on the hub

**Reply by jwnremt** [reply]
if you set your recipies as public you should be able to do http://URL/g/groupname

**Comment by philbar** [comment]
Does anyone know why this request was closed as resolved?

**Reply by Mraedis** [reply]
Yes, the OP did that.
